networkx/generators/community.py

```diff
@@ -406,3 +410,44 @@ def gaussian_random_partition_graph(n, s, v, p_in, p_out, directed=False,
         assigned += size
         sizes.append(size)
     return random_partition_graph(sizes, p_in, p_out, directed, seed)
+
+
+def ring_of_cliques(x, y):
+    """Returns a ring of cliques graph of ``x`` cliques of size ``y``.
+
+    Parameters
+    ----------
+    x : int
+      Number of cliques
+    y : int
+      Size of cliques
+
+    Returns
+    -------
+    G : NetworkX Graph
+      ring of cliques graph
+
+    Raises
+    ------
+    NetworkXError
+        If the number of cliques is lower than 2 or
+        if the size of cliques is smaller than 1.
+
+    Examples
+    --------
+    >>> G = nx.ring_of_cliques(8, 4)
+    """
+    if x < 2:
+        raise nx.NetworkXError('A ring of cliques must have at least '
+                               'two cliques')
+    if y < 1:
+        raise nx.NetworkXError('The cliques must have at least one node')
+
+    G = nx.Graph()
+    for i in range(0, x):
+        edges = itertools.combinations(range(i*y, i*y+y), 2)
+        G.add_edges_from(edges)
+        # connects the middle node of each clique
```

Comment:
Oh so it will look like there is a circle passing through the center of the cliques, as opposed to cliques attached tangentially to an inner circle? Might be good to provide that reasoning in a comment.